### PR TITLE
created dockerfile + added api-service to docker-compose.yml

### LINKED ISSUE #16: Add Dockerfile and service to docker-compose for API service
We need to bringup our api-service as a contianer, and we should be able to query it for saving and getting data.
so pls
design Dockerfile for api service
add entry to docker compose
check that port is available and test with curl

### PR DESCRIPTION
**Pull request by DmitrySavinskikh**
I don't know how to run this docker container and has this code some mistakes?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM python
+
+WORKDIR /opt/service
+
+RUN pip install poetry
```
Comment (EgorBlagov) [inline Dockerfile]: let's specify poetry version here (take from Makefile)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM python
+
+WORKDIR /opt/service
+
+RUN pip install poetry
+RUN poetry install
```
Comment (EgorBlagov) [inline Dockerfile]: Poetry cannot install anything (there is no pyproject.toml and poetry.lock at this step)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
+FROM python
+
+WORKDIR /opt/service
+
+RUN pip install poetry
+RUN poetry install
+RUN uvicorn comprehensive_microservice.api_service:app
```
Comment (EgorBlagov) [inline Dockerfile]: you cannot RUN uvicorn inside the docker file
let's specify uvicorn run command in the docker-compose.yml file (google docker compose service command override)

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+WORKDIR /opt/service
+
+RUN pip install poetry
+RUN poetry install
+RUN uvicorn comprehensive_microservice.api_service:app
+
+COPY . .
+
+ENV NAME venv
```
Comment (EgorBlagov) [inline Dockerfile]: why we need that?

File: docker-compose.yml [hunk 5]
```diff
@@ -1,9 +1,13 @@
...
     image: postgres:14.1-alpine
     environment:
       - POSTGRES_USER=postgres
       - POSTGRES_PASSWORD=postgres
     ports:
-      - '5432:5432'
\ No newline at end of file
+      - '5432:5432'
+  api-service:
+    build: .
```
Comment (EgorBlagov) [inline docker-compose.yml]: as I mentioned, you can specify `command` option here

File: .dockerignore [hunk 6]
```diff
@@ -0,0 +1 @@
+*venv
```
Comment (EgorBlagov) [inline .dockerignore]: why not `.venv`?

File: Dockerfile (hunk 2 above)
Comment (EgorBlagov) [reply Dockerfile]: Also, let's make poetry not create virtualenv inside our docker image

File: Dockerfile (hunk 4 above)
Comment (DmitrySavinskikh) [reply Dockerfile]: to create venv in docker image. but probably it doesn't matter

File: .dockerignore (hunk 6 above)
Comment (DmitrySavinskikh) [reply .dockerignore]: Oh, yes, thank you

File: Dockerfile [hunk 7]
```diff
@@ -2,10 +2,12 @@ FROM python
...
+
+RUN pip install poetry==1.5.1
+RUN poetry config installer.max-workers 10
 RUN poetry install
-RUN uvicorn comprehensive_microservice.api_service:app
 
 COPY . .
 
-ENV NAME venv
+CMD ["uvicorn", "app.main:app", "--host", "0.0.0.0", "--port", "80"]
```
Comment (EgorBlagov) [inline Dockerfile]: this line seems invalid

File: Dockerfile (hunk 2 above)
Comment (EgorBlagov) [reply Dockerfile]: virtualenv thing stil was not solved

File: docker-compose.yml [hunk 8]
```diff
@@ -11,3 +11,7 @@ services:
     build: .
     ports:
       - "8080:80"
+  uvicorn:
+    image: uvicorn comprehensive_microservice.api_service:app
+    ports:
+      - "8000:8000"
```
Comment (EgorBlagov) [inline docker-compose.yml]: what is that?

File: Dockerfile (hunk 7 above)
Comment (EgorBlagov) [reply Dockerfile]: move it to docker-compose to the `command` section as I mentioned, and fix pls

File: Dockerfile (hunk 4 above)
Comment (EgorBlagov) [reply Dockerfile]: you just specify environment variable named NAME and assing 'venv' to it. How does it help to create virtual env?

**Comment by EgorBlagov** [comment]
the problem with "/opt/service/comprehensive_microservice does not contain any element" can be fixed, check this stackoverflow thread: https://stackoverflow.com/questions/75397736/poetry-install-on-an-existing-project-error-does-not-contain-any-element

**Review by EgorBlagov — APPROVED** [review]

**Pull request merged**